packages/mermaid/src/diagrams/xychart/parser/xychart.jison

```diff
@@ -0,0 +1,201 @@
+%lex
+%options case-insensitive
+
+%x string
+%x md_string
+%x title
+%x open_directive
+%x type_directive
+%x arg_directive
+%x close_directive
+%x acc_title
+%x acc_descr
+%x acc_descr_multiline
+%s axis_data
+%s data
+%s data_inner
+%%
+\%\%\{                                    { this.begin('open_directive'); return 'open_directive'; }
+<open_directive>((?:(?!\}\%\%)[^:.])*)    { this.begin('type_directive'); return 'type_directive'; }
+<type_directive>":"                       { this.popState(); this.begin('arg_directive'); return ':'; }
+<type_directive,arg_directive>\}\%\%      { this.popState(); this.popState(); return 'close_directive'; }
+<arg_directive>((?:(?!\}\%\%).|\n)*)      return 'arg_directive';
+\%\%(?!\{)[^\n]*                          /* skip comments */
+[^\}]\%\%[^\n]*                           /* skip comments */
+<axis_data>(\r?\n)                        { this.popState(); return 'NEWLINE'; }
+<data>(\r?\n)                             { this.popState(); return 'NEWLINE'; }
+[\n\r]+                                   return 'NEWLINE';
+\%\%[^\n]*                                /* do nothing */
+
+title                                     { this.begin("title"); return 'title'; }
+<title>(?!\n|;|#)*[^\n]*                  { this.popState(); return "title_value"; }
+
+
+accTitle\s*":"\s*                         { this.begin("acc_title");return 'acc_title'; }
+<acc_title>(?!\n|;|#)*[^\n]*              { this.popState(); return "acc_title_value"; }
+accDescr\s*":"\s*                         { this.begin("acc_descr");return 'acc_descr'; }
+<acc_descr>(?!\n|;|#)*[^\n]*              { this.popState(); return "acc_descr_value"; }
+accDescr\s*"{"\s*                         { this.begin("acc_descr_multiline");}
+<acc_descr_multiline>[\}]                 { this.popState(); }
+<acc_descr_multiline>[^\}]*               { return "acc_descr_multiline_value"; }
+
+"xychart-beta"		                        {return 'XYCHART';}
+("vertical"|"horizontal")                 {return 'CHART_ORIENTATION'}
+
+"x-axis"                                  { this.begin("axis_data"); return "X_AXIS"; }
+"y-axis"                                  { this.begin("axis_data"); return "Y_AXIS"; }
+<axis_data>[\[]                           { this.popState(); return 'SQUARE_BRACES_START'; }
+<axis_data>[+-]?\d+(?:\.\d+)?             { return 'NUMBER_WITH_DECIMAL'; }
```

Comment:
@sidharthv96 @nirname Do we want to consider including numbers that omit the first zero for fractional numbers, i.e., allowing things like `.45`, `+.12`, and `-.15`. I can imagine in the future someone complaining that needing a zero (or any number, really) on an axis that ranges from 0 to 1 would be annoying.

Reply:
I am ok with both. Many languages support omitting leading zero so we could support this as well. But not necessarily.

Reply:
@ibrahimWassouf @nirname @sidharthv96 added support for this also

Reply:
@subhash-halder good to hear!